### PR TITLE
[WIP: CodeCov patch diff hit failure] CI: add source directory for logs CI: add source directory for logs

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Meson and Autotools dump configuration files to the source directory.

ref. https://github.com/spack/spack/pull/42213#issuecomment-2040009344

### CODE REVIEW

File: lib/spack/spack/build_systems/autotools.py [hunk 1]
```diff
@@ -187,7 +187,11 @@ def _removed_la_files_log(self):
     @property
     def archive_files(self):
         """Files to archive for packages based on autotools"""
-        files = [os.path.join(self.build_directory, "config.log")]
+        archives = ("config.log", "config.guess", "configure")
+        staging_prefixes = (self.build_directory, self.pkg.stage.source_path)
```
Comment (adamjstewart) [inline lib/spack/spack/build_systems/autotools.py:191]: Why do we have to check both? Won't the log always be in the same directory?

**Review by adamjstewart — COMMENTED** [review]
Note that numpy/scipy/etc. are not MesonPackages, they define this property themselves.

File: lib/spack/spack/build_systems/autotools.py (hunk 1 above)
Comment (trws) [reply lib/spack/spack/build_systems/autotools.py:191]: In principle we shouldn't have to for config.log, if build_directory always points to the directory that's actually configured, but usually configure and config.guess will live only in the source tree, and sometimes config.log seems to elude us only looking in the build directory.

**Review by trws — APPROVED** [review]

**Comment by adamjstewart** [comment]
Can you also check other files that manually set this variable?
```console
> spack pkg grep archive_files
/Users/Adam/spack/var/spack/repos/builtin/packages/cp2k/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/petsc/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/py-matplotlib/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/py-numpy/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/py-scipy/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/qt-base/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/slepc/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/spectre/package.py:    def archive_files(self):
/Users/Adam/spack/var/spack/repos/builtin/packages/spectre/package.py:        return super().archive_files + [join_path(self.build_directory, "BuildInfo.txt")]
```
I mostly care about the Python ones, but there aren't too many so might as well check all. This whole thing was prompted by scipy, which currently isn't covered by this PR.

**Comment by trws** [comment]
Do you have a link to what you mean @adamjstewart? I became aware of this through a different path, build issue on GasNet that didn't pop out a config.log artifact, so I haven't seen that one.

**Comment by adamjstewart** [comment]
See https://github.com/spack/spack/pull/42213#issuecomment-2039919820

**Comment by kwryankrattiger** [comment]
@adamjstewart thanks for grabbing that list! I looked at those packages and they all seem to either be using `super().archive_files` correctly, or are implementing their own config/build for the package.

**Comment by adamjstewart** [comment]
@kwryankrattiger scipy does not use `super` and is not fixed by this PR.

**Comment by kwryankrattiger** [comment]
Ah right sorry. I am building that locally to get a better idea of what is going on there.

**Comment by kwryankrattiger** [comment]
py-scipy has the meson-log.txt in it as far as I can tell.

**Comment by adamjstewart** [comment]
Hmm, then I'm confused what initially triggered this PR. Is the issue that they aren't being installed to the CI artifact?

**Comment by kwryankrattiger** [comment]
>  I'm confused what initially triggered this PR

There was still an issue with the `config.log` not being present in the logs not being present in `autotools` projects. The `py-scipy` project uses `meson` to build, which was a different issue.

**Comment by scottwittenburg** [comment]
@spackbot run pipeline

**Review by bernhardkaindl — COMMENTED** [review]
@kwryankrattiger: On "CI: add source directory for logs:"

Adding [WIP: CodeCov patch diff hit failure]

Should this PR continue? / Would you add python tests to add test coverage to the changed lines to pass the diff coverage test?

**Comment by kwryankrattiger** [comment]
I don't think there has ever a strong consensus that the original reported problem was actually a problem at all and/or sufficiently solved by this change, which has been the main blocker on this. I am going to close it for now.

**Pull request closed without merging**